### ISSUE TITLE
Plan for ci/infra/release

### ISSUE THREAD

**Issue by srikanthccv** [open-telemetry/opentelemetry-sqlcommenter#5]
I am starting this discussion to consolidate our thoughts/ideas and prepare a plan ahead. I have a couple of questions in my mind that I think should be brought up. Do we eventually migrate all components from this polyglot repo to their respective SIGs main/contrib repos or does this remain as a stand alone project? (I am inclined towards for moving pieces slowly to main/contrib repos). Or is it a mix of both where certain parts remain in this and everything else will be integrated with instrumentations? If we have all or some parts of Sqlcommenter that lives in this repo we will have to take care of bunch of things including CI, Clear contribution guidelines, and release management. Alternatively we can make use already available infra from language SIGs. These are some things on top of my mind. Please share their thoughts and suggestions.

**Comment by sjs994** [comment]
Thanks @lonewolf3739 for starting the discussion thread. All these questions are quite important and needs some good discussion.

This is the initial discussion that happened regarding bringing in SQLCommenter  into OpenTelemetry: https://github.com/open-telemetry/community/issues/783. 

I am new to OpenTelemetry. So, maybe will be missing few complexities the OpenTelemetry project might have. Will try to summarise my thought process.

I am more inclined towards having certain parts integrated into OTel and certain parts be in an independent repository. 
1. The instrumentation part seems to be better if integrated into respective SIGs. That would help in seamless integration without adding extra code to initialise SQLCommenter. There can be configurations to control whether the Database queries will be instrumented or not.
2. Now, regarding the exporter part, maybe we should separate it.
      - The reason being there might be multiple (in-process/out-of-process) plugins/extensions which can be created to export metrics/tracing information from each of the DB engines. Adding them into OpenTelemetry would really bloat it up. 
      - Also from my limited experience database code have their own set of complexities and may need a different set of testing mechanism, failure handling, handling performance based issues, etc.

**Comment by lonewolf3739** [comment]
> I am more inclined towards having certain parts integrated into OTel and certain parts be in an independent repository.

What parts do you think should be reside in this independent repository?

I will take python as an example to share my thoughts and believe same applies to other languages. As you rightly pointed out, instrumentation functionality is better integrated directly with instrumentations provided by opentelemetry. This leaves us with only few parts such as `get_flask_info`, `get_opentelemetry_values`, and `generate_sql_comment`. IMO it is an overkill to maintain them separately. In contrib repos we have separate package where we maintain cross instrumentation helpers, which I think is perfect place to move these definitions. 

> Now, regarding the exporter part, maybe we should separate it.

I am not quite sure I understand this part. Can you elaborate where do you think exporters come into picture?

**Comment by sjs994** [comment]
@aabmass I agree that sqlcommenter don't need opentelemetry, but as @lonewolf3739 made a valid point that its living inside opentelemetry org and logically opentelemetry should be a first party support. I am inclined towards this point.

So, as a start maybe we should aim to make it compliant with opentelemetry only. That will give more traction to the feature/database instrumentation and provide better observability to users. @aabmass what are your thoughts on this ?
